src/common/ceph_context.cc

```diff
@@ -425,7 +426,45 @@ void CephContext::do_command(std::string command, cmdmap_t& cmdmap,
         f->dump_string("option", *p);
       }
       f->close_section(); // unknown
-    } else if (command == "log flush") {
+    }
+    else if (command == "config diff get") {
+      std::string ceph_setting;
+      if (!cmd_getval(this, cmdmap, "var", ceph_setting)) {
+        f->dump_string("error", "syntax error: 'config diff get <var>'");
+      } //-- if (!cmd_getval(this, cmdmap, "var", ceph_setting))
+      else {
+        md_config_t def_conf;
+        def_conf.set_val("cluster", _conf->cluster);
+        def_conf.name = _conf->name;
+        def_conf.set_val("host", _conf->host);
+        def_conf.apply_changes(NULL);
+
+        map<string, pair<string, string> > diff;
+        set<string> unknown;
+        def_conf.diff_setting(_conf, &diff, &unknown, ceph_setting, true);
+        f->open_object_section("diff");
+        f->open_object_section("current");
+
+        for (map<string, pair<string, string> >::iterator p = diff.begin(); p != diff.end(); ++p) {
+          f->dump_string(p->first.c_str(), p->second.second);
+        } //-- for (map<string,pair<string,string> >::iterator p = diff.begin(); p != diff.end(); ++p)
+        f->close_section();   //-- current
+
+        f->open_object_section("defaults");
+        for (map<string, pair<string, string> >::iterator p = diff.begin(); p != diff.end(); ++p) {
+          f->dump_string(p->first.c_str(), p->second.first);
+        } //-- for (map<string,pair<string,string> >::iterator p = diff.begin(); p != diff.end(); ++p)
+        f->close_section();   //-- defaults
+        f->close_section();   //-- diff
+
+        f->open_array_section("unknown");
+        for (set<string>::iterator p = unknown.begin(); p != unknown.end(); ++p) {
+          f->dump_string("option", *p);
+        } //-- for (set<string>::iterator p = unknown.begin(); p != unknown.end(); ++p)
+        f->close_section();   //-- unknown
+      } //-- else if (!cmd_getval(this, cmdmap, "var", ceph_setting))
+    } //-- else if (command == "config diff get")
+  else if (command == "log flush") {
```

Comment:
We already had some misaligned block, so I ended up fixing some up the function in order to see this one properly.